server/src/main/java/org/opensearch/extensions/RegisterRestActionsRequest.java

```diff
@@ -24,49 +26,57 @@
  */
 public class RegisterRestActionsRequest extends TransportRequest {
```

Comment:
I think we would want similar treatment on any request from OpenSearch -> Extension interactions - should a there be a base class for all interactions, or a common wrapper?  @saratvemulapalli have you had thoughts about this space?

Making this more concrete, lets consider the IndicesModuleRequest.

IndicesModuleRequest is translated by ExtensionsOrchestrator to send the request to the Extension (ExtensionRunner).  When the call is send off to the extension, the identity of the operation should be included.  Attaching this information onto of the `Request` object seems most straight forward, but is there another way that would better integrate it?

This would also mean that before transiting the OpenSearch -> Extension boundary there would be some kind of 'get current identity and associate it as a Principle with outbound request' component.

Reply:
Absolutely, as we are working through the security features I think we should start implementing generic message handler which mandates including node identity, security identity etc. 
May be lets start opening up an issue ?